src/js/TuningSliders.js

```diff
@@ -414,61 +449,97 @@ TuningSliders.calculateNewPids = function() {
     FC.FILTER_CONFIG.gyro_lowpass_dyn_min_hz = Math.floor(this.FILTER_DEFAULT.gyro_lowpass_dyn_min_hz * this.sliderGyroFilterMultiplier);
     FC.FILTER_CONFIG.gyro_lowpass_dyn_max_hz = Math.floor(this.FILTER_DEFAULT.gyro_lowpass_dyn_max_hz * this.sliderGyroFilterMultiplier);
     FC.FILTER_CONFIG.gyro_lowpass2_hz = Math.floor(this.FILTER_DEFAULT.gyro_lowpass2_hz * this.sliderGyroFilterMultiplier);
     FC.FILTER_CONFIG.gyro_lowpass_type = this.FILTER_DEFAULT.gyro_lowpass_type;
     FC.FILTER_CONFIG.gyro_lowpass2_type = this.FILTER_DEFAULT.gyro_lowpass2_type;
 
-    $('.pid_filter input[name="gyroLowpassDynMinFrequency"]').val(FC.FILTER_CONFIG.gyro_lowpass_dyn_min_hz);
-    $('.pid_filter input[name="gyroLowpassDynMaxFrequency"]').val(FC.FILTER_CONFIG.gyro_lowpass_dyn_max_hz);
-    $('.pid_filter input[name="gyroLowpass2Frequency"]').val(FC.FILTER_CONFIG.gyro_lowpass2_hz);
-    $('.pid_filter select[name="gyroLowpassDynType').val(FC.FILTER_CONFIG.gyro_lowpass_type);
-    $('.pid_filter select[name="gyroLowpass2Type').val(FC.FILTER_CONFIG.gyro_lowpass2_type);
-    $('output[name="sliderGyroFilterMultiplier-number"]').val(this.sliderGyroFilterMultiplier);
+    this.updateLowpassValues();
 };
 
-TuningSliders.calculateNewDTermFilters = function() {
+TuningSliders.calculateLegacyDTermFilters = function() {
     // calculate, set and display new values in forms based on slider position
     FC.FILTER_CONFIG.dterm_lowpass_dyn_min_hz = Math.floor(this.FILTER_DEFAULT.dterm_lowpass_dyn_min_hz * this.sliderDTermFilterMultiplier);
     FC.FILTER_CONFIG.dterm_lowpass_dyn_max_hz = Math.floor(this.FILTER_DEFAULT.dterm_lowpass_dyn_max_hz * this.sliderDTermFilterMultiplier);
     FC.FILTER_CONFIG.dterm_lowpass2_hz = Math.floor(this.FILTER_DEFAULT.dterm_lowpass2_hz * this.sliderDTermFilterMultiplier);
     FC.FILTER_CONFIG.dterm_lowpass_type = this.FILTER_DEFAULT.dterm_lowpass_type;
     FC.FILTER_CONFIG.dterm_lowpass2_type = this.FILTER_DEFAULT.dterm_lowpass2_type;
 
+    this.updateLowpassValues();
+};
+
+TuningSliders.calculateNewGyroFilters = function() {
+    // this is the main calculation for Gyro Filter slider, inputs are in form of slider position values
+    // values get set both into forms and their respective variables
+    if (semver.lte(FC.CONFIG.apiVersion, API_VERSION_1_43)) {
+        this.calculateLegacyGyroFilters();
+
+    }
+
+    if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) {
+        FC.TUNING_SLIDERS.slider_gyro_filter_multiplier = this.sliderGyroFilterMultiplier * 100;
+
+        Promise.resolve(true)
+        .then(() => { return MSP.promise(MSPCodes.MSP_SET_TUNING_SLIDERS, mspHelper.crunch(MSPCodes.MSP_SET_TUNING_SLIDERS)); })
+        .then(() => { return MSP.send_message(MSPCodes.MSP_FILTER_CONFIG); });
+
+        const delayInMilliseconds = 250; //0.25 seconds
+        setTimeout(function() {
+            TuningSliders.updateLowpassValues();
+        }, delayInMilliseconds);
+    }
```

Comment:
There is not point of doing `Promise.resolve(true)` here,`MSP.promise` already returns a promise. Also why that timeout? Can't we update low pass values after the last promise resolves?

Something like?
```suggestion
        const delayInMilliseconds = 250; //0.25 seconds
        MSP.promise(MSPCodes.MSP_SET_TUNING_SLIDERS, mspHelper.crunch(MSPCodes.MSP_SET_TUNING_SLIDERS))
        .then(() => MSP.send_message(MSPCodes.MSP_FILTER_CONFIG))
        .then(() => TuningSliders.updateLowpassValues())
    }
```

Reply:
@chmelevskij , as outlined below - which is a little simpler to demonstrate the lag - if I try to implement with the filter sliders, since the calcs take 1 to 3.5ms to complete (D-term filter takes the longest), it will shut off the slider if you call `TuningSliders.updateLowpassValues()` and `TuningSliders.updateFilterSlidersDisplay()` in succession without a Timeout.

![iPJu2nEYd3](https://user-images.githubusercontent.com/25570978/109498089-fae9af80-7a60-11eb-82ce-78dc5aca55f1.gif)

Reply:
`send_message`  doesn't return the promise though, so even though you use `then` it resolves straight away. Can you use `.promise` for that as well?

You don't need `Promise.resolve(true)` though ,with or without the timeout

Reply:
@chmelevskij , the promise seems to work well to avoid the time-out to call one function after MPS, thank you!
However, calling two functions still seems to glitch out.  Any suggestions?

![image](https://user-images.githubusercontent.com/25570978/109525858-f254a180-7a7f-11eb-85d6-d4bb9063aa95.png)

![fretwUlIp6](https://user-images.githubusercontent.com/25570978/109526287-64c58180-7a80-11eb-8efb-f355e825f460.gif)

Reply:
Do you get the same if you put both functions in the same promise resolve?

something like:
```
MSP.promise(MSPCodes.MSP_SET_TUNING_SLIDERS, mspHelper.crunch(MSPCodes.MSP_SET_TUNING_SLIDERS))
  .then(() => MSP.promise(MSPCodes.MSP_FILTER_CONFIG)
  .then(() => {
    TuningSliders.updateLowpassValues();
    TuningSliders.updateFiltersSlidersDisplay();
  })
```

Reply:
@chmelevskij that worked!!  Thanks again.  Now the only glitch is you have to press Enables Sliders twice if you turn off a lowpass or adjust a value manually and Save.

Updated code is committed.  
![8NZKYj8XqU](https://user-images.githubusercontent.com/25570978/109586430-3329d600-7ad3-11eb-993d-2fc8ee833b89.gif)